modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1678,7 +1872,10 @@ public IgniteFutureImpl<Void> restoreSnapshot(String name, @Nullable String snpP
         for (File tmp : snapshotTmpDir().listFiles())
             U.delete(tmp);
 
-        deleteSnapshot(snpDir, pdsSettings.folderName());
+        if (INC_SNP_NAME_PATTERN.matcher(snpDir.getName()).matches())
```

Comment:
Looks unreliable, as user might create an ordinary snapshot with a such name. Let's check directory name also (it must contain "increments")

Reply:
Good catch. Check added.